.github/workflows/test.yml

```diff
@@ -3,41 +3,137 @@ name: CI
+                  echo "$HOME/bin" >> $GITHUB_PATH
+                  # TODO for native-comp: export LDFLAGS="-L/usr/local/Cellar/libgccjit/10.2.0_1/lib/gcc/10"
+
+                  ## nproc for mac
+                  # https://github.com/memkind/memkind/issues/33
+                  alias nproc="sysctl -n hw.logicalcpu"
+
+            - name: rustfmt && common env set
               run: |
                   # Configure $PATH: Executables are installed to $HOME/bin
                   export PATH="$HOME/bin:$PATH"
 
-                  rustup install $(cat rust-toolchain)
-                  rustup component add rustfmt-preview
                   ./autogen.sh && ./configure --without-makeinfo --with-x=no --with-ns=no --without-gconf --without-gsettings
                   admin/format-rust.sh
 
+                  ## Set env
+                  # Path Set
+                  echo "/usr/lib/ccache:/usr/local/opt/ccache/libexec" >> $GITHUB_PATH
+
+                  # Compile with clang
+                  echo "CC=clang" >> $GITHUB_ENV
+                  echo "CXX=clang++" >> $GITHUB_ENV
+                  echo "CPP=clang -E" >> $GITHUB_ENV
+                  echo "CFLAGS= -g1 -O2" >> $GITHUB_ENV
+                  echo "CXXFLAGS= -g1 -O2" >> $GITHUB_ENV
+                  echo "AR=llvm-ar" >> $GITHUB_ENV
+                  echo "AS=llvm-as" >> $GITHUB_ENV
+                  echo "RANLIB=llvm-ranlib" >> $GITHUB_ENV
+
+                  # Other Flags
+                  echo "CARGO_INCREMENTAL=0" >> $GITHUB_ENV
+                  echo "RUSTFLAGS=-Zshare-generics=y -Cdebuginfo=1 -Copt-level=0" >> $GITHUB_ENV
+
+            - name: env(for linux)
+              if: runner.os == 'Linux'
+              run: |
+                  echo "LD=lld" >> $GITHUB_ENV
+                  echo "CFLAGS=$CFLAGS -fuse-ld=lld" >> $GITHUB_ENV
+                  echo "CXXFLAGS=$CXXFLAGS -fuse-ld=lld" >> $GITHUB_ENV
+                  echo "RUSTFLAGS=$RUSTFLAGS -Clink-arg=-fuse-ld=lld" >> $GITHUB_ENV
+
+            - name: env(for mac)
+              if: runner.os == 'macOS'
+              run: |
+                  echo "CPLUS_INCLUD=$(/usr/local/opt/llvm/bin/llvm-config --includedir):$CPLUS_INCLUDE_PATH" >> $GITHUB_ENV
+                  echo "LD_LIBRARY_PATH=$(/usr/local/opt/llvm/bin/llvm-config --libdir):$LD_LIBRARY_PATH" >> $GITHUB_ENV
+                  echo "CFLAGS=$CFLAGS -march=native" >> $GITHUB_ENV
+                  echo "CXXFLAGS=$CXXFLAGS -march=native" >> $GITHUB_ENV
+                  echo "RUSTFLAGS=$RUSTFLAGS -Ctarget-cpu=native" >> $GITHUB_ENV
+
+            - name: configure(general)
+              if: ${{ matrix.build == 'general' }}
+              run: |
+                  ./configure --enable-checking=glyphs --enable-rust-debug
+
+            - name: configure(webrender)
+              if: ${{ matrix.build == 'webrender' }}
+              run: |
+                  ./configure --enable-checking=glyphs --enable-rust-debug --with-webrender
```

Comment:
Can we remove `--enable-rust-debug`? I'm trying to change the release pipeline to use the cache as well. However, the debug cache here is not working well with `release` that will be used for creating the release package.
Since it's cache, only the first build is slowed down right?

Reply:
If we use release mode, build will be slower because it uses optimizations like opt-level, lto.
I remember that it was slow even with cache.

Reply:
Build performance and optimization(runtime performance) are trade-offs, so I think it's better to separate them, but it's worth discussing.